index.js

```diff
@@ -25,10 +25,11 @@ var KNOWN_STATICS = {
 };
 
 module.exports = function hoistNonReactStatics(targetComponent, sourceComponent) {
-    var keys = Object.getOwnPropertyNames(sourceComponent);
-    for (var i=0; i<keys.length; ++i) {
-        if (!REACT_STATICS[keys[i]] && !KNOWN_STATICS[keys[i]]) {
-            targetComponent[keys[i]] = sourceComponent[keys[i]];
+    var sourceKeys = Object.getOwnPropertyNames(sourceComponent),
+        targetKeys = Object.getOwnPropertyNames(targetComponent);
+    for (var i=0; i<sourceKeys.length; ++i) {
+        if (!REACT_STATICS[sourceKeys[i]] && !KNOWN_STATICS[sourceKeys[i]] && targetKeys.indexOf(sourceKeys[i]) === -1) {
```

Comment:
Could we use `targetKeys.hasOwnProperty(sourceKeys[i])` here instead of `indexOf`?